### DISCUSSION TITLE
如何设置script可以让oauth2 token 自动的导入在API的header里？

### DISCUSSION THREAD

**Discussion by zhengye1** [dromara/fast-request#308] [General]
我写了一个OAuth2 Auth Server,然后在我的resource server上需要用到我自己这个OAuth2 server产生的token, 但每次都需要copy paste有点麻烦...找了一圈没找到相对应的方法..

以下是我尝试的，但并不成功，我的secure API还是会返回401，估计是没自动导入header去

写了个Pre-request script
···
import cn.hutool.http.HttpRequest
import cn.hutool.http.HttpUtil

HttpRequest request = HttpUtil.createPost("http://localhost:9000/oauth2/token")
Map<String, Object> map = new HashMap<>();
map.put("client_id", "api-client");
map.put("client_secret", "api-secret");
map.put("scope", "apiscope");
map.put("grant_type", "client_credentials");
String result = request.form(map).execute().body();
System.out.println(result)
String token = JSON.parseObject(result).getString("access_token");
request.header("Authorization", "Bearer" +token)
···

**Comment by kings1990** [comment]
[https://api-buddy.cn/guide/script.html](https://api-buddy.cn/guide/script.html)
文档中有介绍到 **request**是内置变量，如果你需要发起别的请求，获取里面的响应，需要注意命名成别的名字，而不是直接用内置变了 request

**Reply by zhengye1** [reply]
懂了，能工作了
